File: intermine/ruby/intermine/test/live_summary_test.rb
require File.dirname(__FILE__) + "/test_helper.rb"

require "test/unit"
require "intermine/service"

class LiveSummaryTest < Test::Unit::TestCase

    def setup
        @service = Service.new("http://localhost/intermine-test")
        @query = @service.query("Employee").where(:age => {:lt => 50})
    end

    def testIteration
        @query.summaries("age").each {|summary|
            assert_equal("37.0243902439024390", summary["average"])
            assert(summary["max"] < 50)
        }

        count = 0
        @query.summaries("name").each {|summary|
            assert_equal(1, summary["count"])
            count += 1
        }
        assert_equal(count, @query.count)

    end

    def testNumericSummary
        summary = @query.summarise("age")
        assert(summary["max"] < 50)
        # Work around float comparison errors, grrr
        assert(summary["average"] > 37.024)
        assert(summary["average"] < 37.025)
    end

    def testNonNumericSummary
        summary = @query.summarise("name")
        assert_equal(1, summary["Tim Canterbury"])
    end

    def testTop
        summary = @query.summaries("department.name")
        top = summary.first
        assert_equal("Warehouse", top["item"])
        assert_equal(12, top["count"])
    end

    def testTemplateSummary
        template = @service.template("CEO_Rivals")
        template_params = {"A" => {"!=" => "Charles Miner"}}

        summary = template.summarise("salary", template_params)
        assert_equal(529119.4, summary["average"])

        summary = template.summarise("company.name", template_params)
        assert_equal(1, summary["Gogirep"])
    end

end

File: intermine/python/intermine-core/tests/live_summary_test.py
import sys
import os
sys.path.insert(0, os.getcwd())

import unittest
from intermine.webservice import Service

class LiveSummaryTest(unittest.TestCase):

    TEST_ROOT = "localhost/intermine-test"
    SERVICE = Service(TEST_ROOT)

    QUERY = SERVICE.select("Employee.*", "department.name")

    def testNumericSummary(self):
        summary = self.QUERY.summarise("age")
        self.assertEqual(10, summary["min"])
        self.assertEqual(74, summary["max"])
        self.assertEqual(44.878787878787875, summary["average"])
        self.assertEqual(12.075481627447155, summary["stdev"])

    def testNonNumericSummary(self):
        summary = self.QUERY.summarise("fullTime")
        self.assertEqual(56, summary[True])
        self.assertEqual(76, summary[False])

        summary = self.QUERY.summarise("department.name")
        self.assertEqual(18, summary["Sales"])

    def testSummaryAsIterator(self):
        path = "department.name"
        q = self.QUERY
        results = q.results(summary_path = path)
        top = results.next()
        self.assertEqual("Accounting", top["item"])
        self.assertEqual(18, top["count"])

        self.assertEqual(top, q.first(summary_path = path))

    def testAliasing(self):
        q = self.QUERY
        self.assertEqual(q.summarise("age"), q.summarize("age"))

if __name__ == '__main__':
    unittest.main()

